src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -250,7 +250,9 @@ private static Project CreateProject(ProjectId projectId, Solution solution)
                 if (docId != null && (projectId == null || docId.ProjectId == projectId))
                 {
                     // does this solution even have the document?
-                    var document = this.GetDocument(docId);
+                    // We call TryGetSourceGeneratedDocumentForAlreadyGeneratedId if it's not a regular document;
+                    // we know that if we already have the tree that means generation completed, so that's safe to call.
+                    var document = this.GetDocument(docId) ?? this.GetProject(docId.ProjectId)?.TryGetSourceGeneratedDocumentForAlreadyGeneratedId(docId);
```

Comment:
so i'm on the fence here.  It looks like we have an API where there is a split on GetDocument(id) vs GetSourceGeneratedDocument(id).  But we don't have the same split for SyntaxTree.  I can't tell if that's good, or confusing.

Reply:
Yeah, agreed. I wish I could make GetDocument(id) work, but alas that isn't async so we run into problems there, whereas this can always be done synchronously regardless. I did a pass through the Roslyn codebase and generally there are a bunch of places that people assume GetDocument() will _never_ return null, and having to update them to a "get the regular or generated document" helper would be an unnecessary change. I'll have a follow up PR after this one where we update a bunch of features and I think that'll be a good place to tweak this if we want to split it, and I think that'll give a much better perspective on how it's generally used across our codebase.